### DISCUSSION TITLE
MG4 GPS Co-ordinates - Zero Value?

### DISCUSSION THREAD

**Discussion by thompct0** [ad-ha/mg-saic-ha#62] [General]
Hi - firstly, great work with the integration, thank you!

Question... the GPS Location reported when the MG4 is powered on (i.e. Driving) is reported ok and Home Assistant locates it correctly as per the zones defined.  Looks like when the car is "off" though that any update, triggered by a manual update of vehicle data,  to the GPS location is then Zero'd out (Assume the car has powered down and is not reporting anything?).

Can you replicate this or just me? :-)

![image](https://github.com/user-attachments/assets/2855ce3b-27bd-4cba-91da-44da18fb5a6a)
![image](https://github.com/user-attachments/assets/14a1bc45-e02b-4d12-ab73-0facc23cbf39)

**Comment by ad-ha** [comment]
Hi @thompct0 

That's quite odd to me. I never add a similar behavior on my Marvel R. Even when the car is off, I still get the GPS coordinates. 

I only received a 0 location twice, during my usage and all my testing, and in both cases I had the car on the garage so I assumed it was an issue with coverage, but never happen again.

**Reply by thompct0** [reply]
Yeah it's a strange one - I did think it could be lack of signal but it happens at multiple locations (only when the car is powered off).  I'll keep an eye on it.

**Comment by ad-ha** [comment]
Let me know if you find something, as I do not get that at all I cannot try to find a fix for it.

Cheers

**Reply by thompct0** [reply]
Hi  - I've extracted the data from HA around the geodata   / No Idea why the car is reporting 0 valued co-ordinates.

Is there any way in the integration logic for it to ignore a 0.0, 0.0 entry and report the last "actual" location (the car is never going to be on Null Island hopefully :-)

**Reply by thompct0** [reply]
![Geo_Data](https://github.com/user-attachments/assets/c69286fe-2d46-443c-9e03-404534abe2d6)

**Reply by ad-ha** [reply]
Hi @thompct0 

Sure, let me have a look how can I handle those 0 values. 

Do you have any logs with the 0 response, so I may have a look into what is actually being sent by the API?

I will try to add an exclusion as a generic response, on the next release.

Cheers

**Reply by thompct0** [reply]
Hi - Great thank you - just did a debug on the integration and triggered an update - rest of the attributes had values populated 

![image](https://github.com/user-attachments/assets/cee53c93-56fe-4680-a7fe-b1720efbf25e).

Unless there is somewhere else or a better way to check this? (Please let me know as I don't have that knowledge).

Cheers

**Reply by ad-ha** [reply]
That looks fine. Just an empty response.

Let me see, whenever I get a moment, how to get it sorted. 

Just spent sometime with the new release and need to work on something else, but I will definitely look into it.

Cheers

**Comment by ad-ha** [comment]
@thompct0 Got it.

I will try to get the last valid coordinates stored and when new data comes in, if equal to 0, will be discarded and last valid data will be shown instead.

This should solve your issue.

Maybe tomorrow I will get it all tested and finalized, so I may try to get it sorted by then.

Cheers

**Comment by ad-ha** [comment]
@thompct0 

Just released version 0.8.1. This should address your issues.

Let me know if it works fine for you.

Cheers
